src/routes/v2/authenticated/sites.ts

```diff
@@ -71,7 +71,26 @@ export class SitesRouter {
     if (possibleStagingUrl instanceof BaseIsomerError) {
       return res.status(404).json({ message: possibleStagingUrl.message })
     }
-    return res.status(200).json({ possibleStagingUrl })
+    return res.status(200).json({ stagingUrl: possibleStagingUrl })
+  }
+
+  getSiteUrl: RequestHandler<
+    { siteName: string },
+    unknown,
+    never,
+    never,
+    { userWithSiteSessionData: UserWithSiteSessionData }
+  > = async (req, res) => {
+    const { userWithSiteSessionData } = res.locals
+    const possibleSiteUrl = await this.sitesService.getSiteUrl(
+      userWithSiteSessionData
+    )
+
+    // Check for error and throw
+    if (possibleSiteUrl instanceof BaseIsomerError) {
+      return res.status(404).json({ message: possibleSiteUrl.message })
+    }
```

Comment:
Could we throw a `NotFoundError` instead so it's easier to standardise our errors?

Reply:
Sadge, I'd actually prefer not to

Reply:
Yeah @seaerchin [mentioned before](https://github.com/isomerpages/isomercms-backend/pull/513#discussion_r994350033) that the error handling middleware should be used for handling unknown exceptions, and we should be able to return the appropriate response for known exceptions.

I don't have too strong opinions but I think it makes sense that we handle errors that we know of here in the router level. It also makes it clear here on what is known to be returned to the user (along with the HTTP status codes). I would see it more like router level handles 4xx errors whereas middleware handles 5xx errors.

Reply:
to add onto this point, we need to separate _exceptions_ from _errors_ . exceptions are things that our application cannot handle and we are forced to rely on a generic error message to show to the user. These are handled by our middleware (and i've introduced a new error type for this on the FE on the collaborators refactor PR).

Conversely, errors are things that we know about and we can return a well-formed message to the user, rather than just falling back on a uninformative message. In this case, we can return a known shape which the frontend can consume readily and provide targeted feedback to the user (eg: no access rights). 

I think a good next step would be to add a util function on the frontend to soothe over this process (i think i added one, but needs to be refined), so that if we get an _expected error_, we use the backend's error message and otherwise, fallback on our generic error messages (ie, your internet slow)